test/test-helpers.coffee

```diff
@@ -143,6 +143,27 @@ beforeEach ->
       @message = -> "Expected to emit #{jasmine.pp(expectedLog)}, actually emitted #{jasmine.pp(log)}"
       @env.equals_(expectedLog, log)
 
+    # Test with setTimeout not guaranteeing order. See
+    # https://github.com/rpominov/kefir/issues/134
+    toEmitOverShakyTime: (expectedLog, cb, timeLimit = 1000) ->
+      log = null
+      exports.withFakeTime (tick) =>
+        originalST = global.setTimeout
+        global.setTimeout = ->
+          if Math.random() > 0.5
```

Comment:
Each run calls setTimeout twice, and I didn't want to couple `toEmitOverShakyTime` with that. I also wanted it to test the regular cases where both setTimeout calls are not or are delayed together, which shouldn't ever fail but act as a good control for the test really just testing `toEmitOverShakyTime` itself.

I've put in another commit that removes the non-determinism and replaces it with code closer to your suggestion.